### DISCUSSION TITLE
Terminal forgets the environment variables of .env file  (if the terminal is killed)

### DISCUSSION THREAD

**Discussion by Prezzy06** [Cyfrin/foundry-full-course-cu#23] [Q&A] [answered]
I am having issue in deploying the scripts of part 1 video.
I have declared the environment variables in `.env` file i.e. `PRIVATE_KEY` and `RPC_URL` and to save those variables , I have run `source .env` .  
But if I kill the terminal and  I again deploy the script contract , it is giving me the following error : 
```bash
prezzy_06@Mortis:~/work/foundry-f23/foundry-simple-storage-f23$ forge script script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL --broadcast --private-key $PRIVATE_KEY
error: a value is required for '--fork-url <URL>' but none was supplied

For more information, try '--help'.
```
But if I again run `source .env`, the error disappears and runs properly.
```bash
prezzy_06@Mortis:~/work/foundry-f23/foundry-simple-storage-f23$ forge script script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL --broadcast --private-key $PRIVATE_KEY
error: a value is required for '--fork-url <URL>' but none was supplied

For more information, try '--help'.
prezzy_06@Mortis:~/work/foundry-f23/foundry-simple-storage-f23$ source .env
prezzy_06@Mortis:~/work/foundry-f23/foundry-simple-storage-f23$ forge script script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL --broadcast --private-key $PRIVATE_KEY
[⠒] Compiling...
No files changed, compilation skipped
Script ran successfully.
```
Can anybody tell me how to permanently fix this error as I don't want to run `source .env` again and again.
Please help me on this issue.

**Comment by cromewar** [comment, marked as answer]
Hello there @Prezzy06 if you setup for env files into your console they are just going to be there as far as you have that terminal opened, foundry relates on  `TOML` files  which unfortunately just support .env variables at runtime, it means just when you have that terminal instance opened as I just said.  This is to how Forge works.

Please check out this: https://github.com/foundry-rs/foundry/issues/1034 here they discuss this problem and end up adding support for .env.
But as mentioned forge does not support `.env` files natively, check this: https://github.com/foundry-rs/foundry/issues/2430#issuecomment-1191575171

So, for now the only way is to use that `source .env` and runtime.

Do you know any workaround for this @PatrickAlphaC ?

**Reply by Prezzy06** [reply]
@cromewar Thanks for replying ser. I was having a hard time searching about this error :) . I read those issues and I think they will probably add this `enhancement` in the future.

**Reply by cromewar** [reply]
Let's hope they do so, specially because Frameworks such as Harhdat or brownie actually do.

**Reply by PatrickAlphaC** [reply]
You got it @Prezzy06! In the setup we talk about this, but it's good to have it again as a discussion.

The better solution is to add your scripts into your `Makefile` which has the `-include .env` and the makfile will always pull in your `.env` that way!

**Reply by Prezzy06** [reply]
Thanks @cromewar @PatrickAlphaC ser for resolving the issue. I'm gonna do it right now!

**Comment by alymurtazamemon** [comment]
@Prezzy06 In part 2 of the video in lesson 7, this issue is discussed in detail, please wait and continue watching.
